### PR TITLE
fix: Router label copy

### PR DESCRIPTION
**Pull request by just-toby**
## Description

fixes a bug in the `RouterLabel` component where we checked an unused field to determine if the routing was client-side. now we correctly use the `QuoteMethod` field. 

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2597/fix-routerlabel-copy-for-local-routing

### Before

image

### After

image

### Reproducing the error

1. load a quote with client side routing forced and notice it still says "API"

### QA (ie manual testing)

- [x] repeat above steps, notice it says "client" instead of "API"

### Automated testing

- [x] Unit test
- [x] Integration/E2E test N/A

### CODE REVIEW

File: src/components/RouterLabel/index.tsx
Comment (gnewfield) [inline src/components/RouterLabel/index.tsx:1]: Is `isUniswapXTrade` function working? When I select UniswapX as my preferred router, I'm still seeing Uniswap API in the label.
Screenshot 2023-08-03 at 4 56 46 PM

Comment (cartcrom) [reply src/components/RouterLabel/index.tsx:1]: You'll only get a UniswapX quote if it's better than a classic trade. You can enable synthetic quotes in the feature flag settings to get them more often

**Review by tinaszheng — APPROVED** [review]
💯

**Review by cartcrom — APPROVED** [review]
🔥

File: src/components/RouterLabel/index.tsx
Comment (gnewfield) [reply src/components/RouterLabel/index.tsx:1]: Nice thanks for clarifying!

**Pull request merged**